### PR TITLE
Fix fax attachments and some others changes

### PR DESCRIPTION
**Pull request by icehess**
* Fix an issue when sending a fax error template with no fax doc or attachment resulted in crash
* Increase timeout for notify resender to 10s to allow templates like voicemail to fetch attachments
* ~~Remove obsolete `register` template~~
* Return and log failure when data is invalid instead of crash so the publisher won't wait to timeout
* See if any templates binding was successful, if not return failure so the publisher won't wait to timeout

### CODE REVIEW

File: applications/teletype/src/teletype_bindings.erl [hunk 1]
```diff
@@ -36,11 +36,22 @@ flush_mod(Module) ->
...
     {EventCategory, EventName} = kz_util:get_event_type(JObj),
     RoutingKey = ?ROUTING_KEY(EventCategory, EventName),
     lager:debug("dispatching notification ~s", [RoutingKey]),
-    _ = kazoo_bindings:map(RoutingKey, JObj, []),
-    'ok'.
+    Res = kazoo_bindings:map(RoutingKey, JObj, []),
+    case kazoo_bindings:succeeded(Res, fun filter_out_failed/1) of
+        [] ->
+            lager:debug("notification ~s did not result in successes", [RoutingKey]),
+            teletype_util:send_update(JObj, <<"failed">>, <<"not_succeeded">>);
```
Comment (fenollp) [inline applications/teletype/src/teletype_bindings.erl]: Is there no way to get more info than failed/not_succeeded? Or is that intended?

Comment (icehess) [inline applications/teletype/src/teletype_bindings.erl]: hehe just saw this, I updated it, take a look.

File: applications/teletype/src/teletype_fax_util.erl [hunk 2]
```diff
@@ -22,129 +20,194 @@
...
-            ,[<<"fax">>, <<"tx_result">>, <<"timezone">>]
-            ,[<<"fax">>, <<"rx_result">>, <<"timezone">>]
-            ],
-    find_timezone(kz_json:get_first_defined(Paths, DataJObj), FaxBoxJObj).
+-spec add_attachments(kz_json:object(), kz_proplist(), boolean()) -> {kz_proplist(), attachments()}.
+add_attachments(DataJObj, Macros, ShouldTerminate) ->
+    IsPreview = teletype_util:is_preview(DataJObj),
+    FaxDoc = props:get_value(<<"fax">>, Macros),
+    case kz_json:is_json_object(FaxDoc)
+        andalso kz_json:is_empty(FaxDoc)
```
Comment (lazedo) [inline applications/teletype/src/teletype_fax_util.erl]: @icehess it seems that if there is a nonempty FaxDoc it will not try to get the attachment ?

Comment (icehess) [inline applications/teletype/src/teletype_fax_util.erl]: yes

**Pull request merged**